broker/cluster.go

```diff
@@ -0,0 +1,170 @@
+				p.instanceMap[instanceID].LastOperation.State = SucceedOperationState
+				p.instanceMap[instanceID].LastOperation.Description = SucceedOperationDescription
+				log.Printf("Starting...[done] %s", okmsg)
+				return nil
+			case omsg := <-msg:
+				switch omsg.(type) {
+				case dbaas.OutuputMsgDebug:
+					// fmt.Printf("\n[debug] %s\n", omsg)
+				case dbaas.OutuputMsgError:
+					p.instanceMap[instanceID].LastOperation.State = FailedOperationState
+					p.instanceMap[instanceID].LastOperation.Description = FailedOperationDescription
+					log.Printf("[operator log error] %s\n", omsg)
+				}
+			case err := <-cerr:
+				switch err.(type) {
+				case dbaas.ErrAlreadyExists:
+					p.instanceMap[instanceID].LastOperation.State = FailedOperationState
+					p.instanceMap[instanceID].LastOperation.Description = InProgressOperationDescription
+					log.Printf("[ERROR] %v", err)
+					list, err := dbaas.List("pxc")
+					if err != nil {
+						return nil
+					}
+					log.Println("Avaliable clusters:")
+					log.Print(list)
+				default:
+					log.Printf("[ERROR] create pxc: %v", err)
+				}
+			}
+		}
+	case psmdbServiseID:
+		app := psmdb.New(instance.Parameters.ClusterName, instance.Parameters.ClusterName, defaultVersion)
+		conf := dbaas.ClusterConfig{}
+		SetDefault(&conf)
+
+		var s3stor *dbaas.BackupStorageSpec
+
+		setupmsg, err := app.Setup(conf, s3stor)
+		if err != nil {
+			log.Println("[Error] set configuration:", err)
+			return nil
+		}
+
+		log.Println(setupmsg)
+
+		created := make(chan string)
+		msg := make(chan dbaas.OutuputMsg)
+		cerr := make(chan error)
+
+		go dbaas.Create("psmdb", app, created, msg, cerr)
+
+		for {
+			select {
+			case okmsg := <-created:
+				p.instanceMap[instanceID].LastOperation.State = SucceedOperationState
+				p.instanceMap[instanceID].LastOperation.Description = SucceedOperationDescription
+				log.Printf("Starting...[done] %s", okmsg)
+				return nil
+			case omsg := <-msg:
+				switch omsg.(type) {
+				case dbaas.OutuputMsgDebug:
+					// fmt.Printf("\n[debug] %s\n", omsg)
+				case dbaas.OutuputMsgError:
+					p.instanceMap[instanceID].LastOperation.State = FailedOperationState
+					p.instanceMap[instanceID].LastOperation.Description = FailedOperationDescription
+					log.Printf("[operator log error] %s\n", omsg)
+				}
+			case err := <-cerr:
+				switch err.(type) {
+				case dbaas.ErrAlreadyExists:
+					p.instanceMap[instanceID].LastOperation.State = FailedOperationState
+					p.instanceMap[instanceID].LastOperation.Description = InProgressOperationDescription
+					log.Printf("[ERROR] %v", err)
+					list, err := dbaas.List("pxc")
+					if err != nil {
+						return nil
+					}
+					log.Println("Avaliable clusters:")
+					log.Print(list)
+				default:
+					log.Printf("[ERROR] create psmdb: %v", err)
+				}
+			}
+		}
+	}
+	return nil
+}
+
+func (p *Controller) DeletePXCCluster(instance *ServiceInstance) error {
+	ok := make(chan string)
+	cerr := make(chan error)
+	delePVC := true
```

Comment:
Should we delete PVC? Should it be an option (can it be)?

Reply:
set to "true"